rdflib/container.py

```diff
@@ -0,0 +1,303 @@
+        
+        container = self.uri
+        for item in other:
+            
+            end+=1
+            self.len+=1
+            elem_uri='http://www.w3.org/1999/02/22-rdf-syntax-ns#_'+str(end)
+            self.graph.add((container, URIRef(elem_uri), item))
+
+
+        
+
+    def clear(self):
+        '''removing all elements from the container'''
+        container = self.uri
+        graph = self.graph
+        i=1
+        while True:
+            elem_uri='http://www.w3.org/1999/02/22-rdf-syntax-ns#_'+str(i)
+            if(container, URIRef(elem_uri), None) in self.graph:
+                graph.remove((container, URIRef(elem_uri), None))
+                i+=1
+            else:
+                break
+        self.len=0
+
+        
+
+class Bag(Container):
+    def __init__(self, graph, uri, seq=[]):
+        Container.__init__(self,graph,uri,seq,"Bag")
+
+class Alt(Container):
+    def __init__(self, graph, uri, seq=[]):
+        Container.__init__(self,graph,uri,seq,"Alt")
+
+    '''returns any one item of the Alt Container randomly'''
+    def anyone(self):
+        if (self.__len__==0):
+            raise  NoElementException()
+        else:
+            p=randint(1,self.__len__())
+            item=self.__getitem__(p)
+            return item
+
+class Seq(Container):
+    def __init__(self, graph, uri, seq=[]):
+        Container.__init__(self,graph,uri,seq,"Seq")
+
+    '''adds the item at position pos, valid values are 1 to len+1'''
+    def add_at_position(self,pos,item):
+        assert isinstance(pos,int)
+        if pos<=0 or pos>(self.__len__()+1):
+            raise ValueError("Invalid Position for inserting element in rdf:seq")
+
+        if (pos==(self.__len__()+1)):
+            self.append(item)
+        else:
+            for j in range(self.__len__(),pos-1,-1):
+                container=self._get_container()
+                elem_uri='http://www.w3.org/1999/02/22-rdf-syntax-ns#_'+str(j)
+                v=self.graph.value(container,URIRef(elem_uri))
+                self.graph.remove((container,URIRef(elem_uri),v))
+                elem_uri='http://www.w3.org/1999/02/22-rdf-syntax-ns#_'+str(j+1)
+                self.graph.add((container,URIRef(elem_uri),v))
+            elem_uri_pos='http://www.w3.org/1999/02/22-rdf-syntax-ns#_'+str(pos)
+            self.graph.add((container,URIRef(elem_uri_pos),item))
+            self.len+=1
+
+
+
+
+
+
+
+
+class NoElementException(Exception):
+    def __init__(message="Alt Container is empty"):
+        self.message=message
+
+    def __str__(self):
+        return message
+
+
+
+
+
+if __name__ == "__main__":
+    #test()
+
+    from rdflib import Graph
+    g = Graph()
+
+    c = Bag(g, BNode())
+
+    assert c.__len__() == 0
+
+    cc = Bag(
+        g, BNode(), [Literal("1"), Literal("2"), Literal("3"), Literal("4")])
+
+    print (cc.items())
+
+    from pprint import pprint
+    pprint (cc.n3())
+    assert cc.__len__() == 4

```

Comment:
Any reason for using the protocol-dunder-methods here instead of using regular syntax (i.e. `len(...)`, `cc[1]`, `del ...` etc.)? Using regular syntax would make the tests represent regular usage and would be more clear IMHO.

Reply:
Hey @niklasl 
I have now used regular syntax for the tests as suggested by you.